docs/extract.go

```diff
@@ -74,22 +76,42 @@ var (
 {{fencedCode .ExampleSource}}
 
 ## Parameters
+
+Here are the HCL fields that you can specify, along with their expected types
+and restrictions:
+
 {{ range .Fields}}
 - {{.Name}} ({{if .Required}}required {{end}}{{if ne .Base ""}}base {{.Base}} {{end}}{{.Type}})
 
 {{ if .MutuallyExclusive}}
-  Only one of {{codeCommaJoin .MutuallyExclusive "or"}} may be set.
+	Only one of {{codeCommaJoin .MutuallyExclusive "or"}} may be set.
 
 {{end}}{{ if .ValidValues}}
-  Valid values: {{codeCommaJoin .ValidValues "and"}}
+	Valid values: {{codeCommaJoin .ValidValues "and"}}
+
+{{end}}{{if ne .Doc ""}}  {{.Doc}}{{end}}
+{{end}}
```

Comment:
this is where that extra line in every doc file is coming from. Would you mind replacing this with `{{- end -}}` to remove the extra whitespace?